File: src/main/java/me/zeroeightsix/kami/module/modules/movement/SafeWalk.java
package me.zeroeightsix.kami.module.modules.movement;

import me.zeroeightsix.kami.module.ModulePlay;

/**
 * Created by 086 on 11/10/2018.
 */
@ModulePlay.Info(name = "SafeWalk", category = ModulePlay.Category.MOVEMENT, description = "Keeps you from walking off edges")
public class SafeWalk extends ModulePlay {

    private static SafeWalk INSTANCE;

    public SafeWalk() {
        INSTANCE = this;
    }

    public static boolean shouldSafewalk() {
        return INSTANCE.isEnabled();
    }

}

File: src/main/java/me/zeroeightsix/kami/feature/module/movement/SafeWalk.kt
package me.zeroeightsix.kami.feature.module.movement

import me.zero.alpine.listener.EventHandler
import me.zero.alpine.listener.EventHook
import me.zero.alpine.listener.Listener
import me.zeroeightsix.kami.event.events.ClipAtLedgeEvent
import me.zeroeightsix.kami.feature.module.Module

/**
 * Created by 086 on 11/10/2018.
 */
@Module.Info(
    name = "SafeWalk",
    category = Module.Category.MOVEMENT,
    description = "Keeps you from walking off edges"
)
object SafeWalk : Module() {

    @EventHandler
    val clipListener = Listener<ClipAtLedgeEvent>(EventHook {
        it.clip = true
    })

}
